packages/amplify-cli/src/utils/global-prefix.ts

```diff
@@ -21,3 +25,62 @@ function getYarnPrefix() {
   }
   return yarnPrefix;
 }
+
+// NOTE: This issue: https://github.com/jonschlinkert/global-prefix/issues/30 prevents us from using the global-prefix package
+function getGlobalNpmPrefix() {
+  // System-wide prefix
+  if (process.env.PREFIX) return process.env.PREFIX;
+
+  // $HOME variable
+  const home = os.homedir();
+  if (home) {
+    const prefix = tryConfigPath(path.resolve(home, '.npmrc'));
+    if (prefix) return prefix;
+  }
+
+  // Find global NPM path
+  const npm = tryWhich('npm');
+  if (npm) {
+    // Check the built-in npm config file
+    const prefix = tryConfigPath(path.resolve(npm, '..', '..', 'npmrc'));
+    if (prefix) {
+      // Custom npm config ?
+      const globalPrefix = tryConfigPath(path.resolve(prefix, 'etc', 'npmrc')) || prefix;
+      if (globalPrefix) return globalPrefix;
+    }
+  }
+
+  const nodePath = tryWhich('node') // Search for Node's global executable
+  if (nodePath) {
+    const { APPDATA, DESTDIR, OSTYPE } = process.env;
+
+    // Windows
+    // c:\node\node.exe --> prefix=c:\node\
+    if (process.platform === 'win32' || OSTYPE === 'msys' || OSTYPE === 'cygwin') {
+      return APPDATA ? path.join(APPDATA, 'npm') : path.dirname(nodePath);
+    }
+  
+    // UNIX
+    // /usr/local/bin/node --> prefix=/usr/local
+    const prefix = path.dirname(path.dirname(nodePath));
+    if (DESTDIR) {
+      return path.join(DESTDIR, prefix);
+    }
```

Comment:
I would try to follow the same pattern that you implemented for windows in line 59 where you check the ```process.platform === linux```

Reply:
Also feel free to elaborate on what your intentions are

Reply:
Do you think the right thing to do is to specifcally check for `macOS || linux`?
I'm afraid to miss some UNIX options that way.

Reply:
The way this is written is correct and in line with the rest of our code where we only special case for Windows.